pkg/credentials/credentials.go

```diff
@@ -234,6 +228,73 @@ func (f *fetcher) parseAWSCredentials() (map[string]string, error) {
 	return creds, nil
 }
 
+func openstackAuthOptionsToCreds(opts *gophercloud.AuthOptions, region string) (map[string]string, error) {
+	creds := make(map[string]string)
+	// We could implement some fallbacks here to support e.g. application credentials / domain ID
+	creds[OpenStackAuthURL] = opts.IdentityEndpoint
+	creds[OpenStackUserNameMC] = opts.Username
+	creds[OpenStackPassword] = opts.Password
+	creds[OpenStackDomainName] = opts.DomainName
+	creds[OpenStackRegionName] = region
+
+	if opts.TenantID != "" {
+		creds[OpenStackTenantID] = opts.TenantID
+	}
+
+	if v, ok := creds[OpenStackTenantID]; !ok || len(v) == 0 {
+		creds[OpenStackTenantName] = opts.TenantName
+	}
```

Comment:
There is a behavior change here that we need to mention in the release note and the docs.

Previously, we just respected the `OS_TENANT_ID` and `OS_TENANT_NAME` environment variables. The library works a bit different as it would set TenantID and TenantName to `OS_PROJECT_ID` and `OS_PROJECT_NAME` values respectively (if those environment variables are present).

> OS_TENANT_ID and OS_TENANT_NAME are deprecated forms of OS_PROJECT_ID and
OS_PROJECT_NAME and the latter are expected against a v3 auth api.

The comment makes it clear and I'm okay with this change, but it should be documented once this PR gets accepted.